api/src/test/java/io/strimzi/api/kafka/model/KafkaTest.java

```diff
@@ -132,6 +137,74 @@ public void testListeners()    {
         assertThat(listeners.get(1).getAuth().getType(), is("tls"));
     }
 
+    @Test
+    public void testListenerTypeAndNameInStatus() throws ParseException, URISyntaxException {
+
+        Kafka kafka = new KafkaBuilder()
+                .withNewMetadata()
+                .withName("my-cluster")
+                .withNamespace("my-namespace")
+                .withGeneration(2L)
+                .endMetadata()
+                .withNewSpec()
+                .withNewKafka()
+                .withReplicas(3)
+                .withListeners(new GenericKafkaListenerBuilder()
+                        .withName("plain")
+                        .withPort(9092)
+                        .withType(KafkaListenerType.INTERNAL)
+                        .withTls(false)
+                        .build())
+                .withNewEphemeralStorage()
+                .endEphemeralStorage()
+                .endKafka()
+                .withNewZookeeper()
+                .withReplicas(3)
+                .withNewEphemeralStorage()
+                .endEphemeralStorage()
+                .endZookeeper()
+                .endSpec()
+                .withNewStatus()
+                .withObservedGeneration(1L)
+                .withConditions(new ConditionBuilder()
+                        .withType("Ready")
+                        .withStatus("True")
+                        .build())
+                .withListeners(new ListenerStatusBuilder()
+                                .withName("plain")
+                                .withAddresses(new ListenerAddressBuilder()
+                                        .withHost("my-service.my-namespace.svc")
+                                        .withPort(9092)
+                                        .build())
+                                .build(),
+                        new ListenerStatusBuilder()
+                                .withName("external")
+                                .withAddresses(new ListenerAddressBuilder()
+                                        .withHost("my-route-address.domain.tld")
+                                        .withPort(443)
+                                        .build())
+                                .build())
+                .endStatus()
+                .build();
```

Comment:
If you indent this along the `withNew...()` and `end...()` methods, it will be much more readable.